novuRajan/gannt-chart
# Backend kickoff

## Description of the problem this PR solves
This PR addresses the need to populate Gantt data dynamically by kicking off an API call to retrieve tasks and subtasks.
## Details of how the solution has been implemented
- Modified the Gantt chart initialization logic to fetch data from the API endpoint.
- Modified the Gantt chart initialization logic to fetch data from the API endpoint.
- Handled cases where the API response format may vary, converting the response to an array when needed.
- If the API call fails, the Gantt chart defaults to using hardcoded data.
## Testing Instructions
1. Run the backend application.
2. Open the Gantt chart web page.
3. Ensure that tasks and subtasks are dynamically fetched from the API and displayed on the Gantt chart.
4. Verify that the Gantt chart gracefully handles API errors by falling back to default hardcoded data.

## Browsers tested
Select all browsers the solution was tested on <br>
*(test on as many as possible if the solution is front-end, otherwise this section is optional)*

## Notes
Any further information required for this PR to be reviewed and tested
